stackslib/src/chainstate/stacks/auth.rs

```diff
@@ -557,14 +763,60 @@ impl TransactionSpendingCondition {
         ))
     }
 
+    pub fn new_multisig_order_independent_p2sh(
+        num_sigs: u16,
+        pubkeys: Vec<StacksPublicKey>,
+    ) -> Option<TransactionSpendingCondition> {
+        let signer_addr = StacksAddress::from_public_keys(
+            0,
+            &AddressHashMode::SerializeP2SH,
+            usize::from(num_sigs),
+            &pubkeys,
+        )?;
+
+        Some(TransactionSpendingCondition::OrderIndependentMultisig(
+            OrderIndependentMultisigSpendingCondition {
+                signer: signer_addr.bytes.clone(),
```

Comment:
Is this `.clone()` needed?

Reply:
Good catch! There are a number of similar functions that I will also fix